File: Module1_Javascript/Lap_trinh_huong_doi_tuong/Exercise/XayDungLopMoTaHinhChuNhat/Rectangle.js
let Rectangle = function (length,width) {
    this.length = length;
    this.width = width;
    this.getLength = function () {
        return this.length;
    }
    this.getWidth = function () {
        return this.width;
    }
    this.setLength = function (length) {
        this.length = length;
    }
    this.setWidth = function (width) {
        this.width = width;
    }
    this.getPerimeter = function () {
        return 2 * (this.length + this.width);
    }
    this.getArea = function () {
        return this.length * this.width;
    }
}

function createRectangle () {
    let c = document.getElementById('canvas').getContext("2d");
    let rectangle = new Rectangle(400,200);
    c.fillStyle = "#FF0000";
    c.fillRect(0,0,rectangle.width,rectangle.length);
    rectangle.setLength(600);
    rectangle.setWidth(300);
    let perimeter1 = rectangle.getPerimeter();
    let area1 = rectangle.getArea();
    alert("Hinh chu nhat co: chieu dai = " + rectangle.getLength() + " - chieu rong = " + rectangle.getWidth() +
        " - chu vi = " + perimeter1 + " - dien tich = " + area1);
}
createRectangle();

File: Module2_Java_core/src/Inheritance/Practice/Shapes/Rectangle.java
package Inheritance.Practice.Shapes;

public class Rectangle extends Shape{
    private double width = 1;
    private double length = 1;

    public Rectangle() {

    }

    public Rectangle(double width, double length) {
        this.width = width;
        this.length = length;
    }

    public Rectangle(double width, double length, String color, boolean filled) {
        super(color, filled);
        this.width = width;
        this.length = length;
    }

    public double getWidth() {
        return width;
    }

    public void setWidth(double width) {
        this.width = width;
    }

    public double getLength() {
        return length;
    }

    public void setLength(double length) {
        this.length = length;
    }

    public double getArea(double width, double length) {
        return width * length;
    }

    public  double getPerimeter(double width, double length) {
        return (2 * (width + length));
    }

    @Override
    public String toString() {
        return "A Rectangle with width="
                + width + " and length="
                + length + ", which is a subclass of "
                + super.toString();
    }
}
